apps/extension/client/src/checkTypegenNesting.ts

```diff
@@ -0,0 +1,57 @@
+import * as vscode from "vscode";
+
+const typeGenPatternKey = "*.ts";
+const typeGenPattern = "${capture}.typegen.ts";
+const hasTypeGenPattern = (fileNestingPatterns: object) =>
+  Object.values(fileNestingPatterns).some((pattern: string) => {
+    if (pattern.includes(",")) {
+      const patterns = pattern.split(",");
+      return patterns.some(
+        (innerPattern) => innerPattern.trim() === typeGenPattern
+      );
+    } else {
+      return pattern === typeGenPattern;
+    }
+  });
+
+export const checkTypegenNestingConfiguration = () => {
+  const fileNestingConfig = vscode.workspace.getConfiguration(
+    "explorer.fileNesting"
+  );
+  const fileNestingPatterns = fileNestingConfig.get<object>("patterns");
```

Comment:
q: does this contain defaults from here?
https://github.com/microsoft/vscode/blob/ecf7f9cfd9ab5460cc2203fb1eb56ae2569157e6/src/vs/workbench/contrib/files/browser/files.contribution.ts#L503-L510

Reply:
We include all the patterns the user already has in there. And it's most likely the defaults, so we keep them and add our needed pattern to the patterns for typescript.